lib/statsd.js

```diff
@@ -464,5 +462,37 @@ Client.prototype.childClient = function(options) {
   return new ChildClient(this, options);
 };
 
+// Overrides tags in parent with tags from child with the same name (case sensitive) and return the result as new
+// array. parent and child are not mutated.
+function overrideTags (parent, child) {
+  var childCopy = {};
+  var toAppend = [];
+  child.forEach(function (tag) {
+    var idx = tag.indexOf(':');
```

Comment:
Joyent's article, in my opinion, is saying that you should not continue after having an unhandled exception.  I very much agree with that in how Node.js works.  I don't agree though that you should allow unhandled exceptions to happen if there is a way to prevent them.  This causes a lot of requests/activities to stop their work potentially and can be a confusing thing to deal with.  

I am fine with this being one of the error cases where we throw an Error (or use the error handler) but would really rather not have it throw an unexpected error.  Especially in the send() function for existing merging of tags, where this is changing existing programs that use them and could suddenly throw uncaught exceptions.